distributed-counter/src/main/java/org/infinispan/counter/CounterListenerCloseable.java

```diff
@@ -0,0 +1,16 @@
+package org.infinispan.counter;
+
+/**
+ * As a return of {@link Counter#addListener(CounterListener)}, it is used to un-register the {@link CounterListener}
+ * via {@link CounterListenerCloseable#close()} method.
+ *
+ * @author Pedro Ruivo
+ * @since 9.0
+ */
+public interface CounterListenerCloseable<T extends CounterListener> extends AutoCloseable {
```

Comment:
I would rather call that `CounterListenerHandle` or something like that (maybe generic `Handle` would do where `CounterListener.addListener` would return `Handle<CounterListener>`. This sounds almost like CloseableCounterListener which would suggest that it's the listener & closeable.